code/modules/reagents/reagents/medical.dm

```diff
@@ -774,39 +768,56 @@
 	name = "Quick Clot Plus"
 	description = "A chemical designed to quickly and painfully remove internal bleeding by encouraging coagulation. Should not be self-administered."
 	color = "#CC00FF"
-	overdose_threshold = REAGENTS_OVERDOSE/5 //Was 4, now 6 //Now 15 //Now 6 again, yay oldQC
-	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5 //10u
+	overdose_threshold = REAGENTS_OVERDOSE/5 //6u
+	overdose_crit_threshold = REAGENTS_OVERDOSE_CRITICAL/5 //12u
 	scannable = TRUE
 	custom_metabolism = REAGENTS_METABOLISM * 2.5
+	///The IB wound this dose of QCP will cure, if it lasts long enough
+	var/datum/wound/internal_bleeding/target_IB
 
 /datum/reagent/medicine/quickclotplus/on_mob_add(mob/living/L, metabolism)
-	if(TIMER_COOLDOWN_CHECK(L, name))
+	if(!ishuman(L))
 		return
-	L.adjustCloneLoss(5*effect_str)
+	var/mob/living/carbon/human/body = L
+	for(var/datum/limb/possible_limb in body.limbs)
```

Comment:
As in for both loops

Reply:
Second one needs to be that way because it's looking specifically for IB and not just any wounds, but you're right on the first